Steps4Impact/NotificationsV2/NotificationsViewController.swift

```diff
@@ -31,35 +31,96 @@ import UIKit
 import RxSwift
 
 class NotificationsViewController: TableViewController {
+  
   override func commonInit() {
     super.commonInit()
 
     title = Strings.Notifications.title
     dataSource = NotificationsDataSource()
   }
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+    checkNotificationPermission()
+  }
+  
+  override func reload() {
+    dataSource?.reload { [weak self] in
+      self?.tableView.reloadOnMain()
+    }
+  }
+  
+  override func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+    super.tableView(tableView, willDisplay: cell, forRowAt: indexPath)
+    if let cell = cell as? NotificationPermissionCell {
+      cell.delegate = self
+    }
+  }
+  
+  func checkNotificationPermission() {
+    func reloadIfNeeded() {
+      guard let dataSource = dataSource as? NotificationsDataSource else {
+        return
+      }
+      if let _ = dataSource.cells.first as? [NotificationPermissionCellContext] {
```

Comment:
This is a confusing part of the MVVM naming scheme for the base `TableViewDataSource` protocol. The `cells` property is of the type `[[CellContext]]` since table views can be displayed as sections of cells.